eslint.config.mjs

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import prettierConfig from 'eslint-config-prettier';
+import config from '@zendeskgarden/eslint-config';
+import noticePlugin from '@zendeskgarden/eslint-config/plugins/notice.js';
+import reactPlugin from '@zendeskgarden/eslint-config/plugins/react.js';
+import typescriptPlugin from '@zendeskgarden/eslint-config/plugins/typescript.js';
+import typescriptTypeCheckedPlugin from '@zendeskgarden/eslint-config/plugins/typescript-type-checked.js';
+import jestPlugin from '@zendeskgarden/eslint-config/plugins/jest.js';
+
+export default [
+  ...config,
+  noticePlugin,
+  prettierConfig,
+  {
+    ignores: ['**/dist']
+  },
+  {
+    rules: {
+      'sort-imports': 'off'
+    }
+  },
+  {
+    ...reactPlugin,
+    rules: {
+      ...reactPlugin.rules,
+      'react/jsx-no-useless-fragment': 'off'
+    }
+  },
+  {
+    files: ['src/**/*.ts'],
+    ...typescriptPlugin,
+    ...typescriptTypeCheckedPlugin,
+    rules: {
+      ...typescriptPlugin.rules,
+      ...typescriptTypeCheckedPlugin.rules,
+      '@typescript-eslint/explicit-function-return-type': 'off',
+      '@typescript-eslint/explicit-module-boundary-types': 'off',
+      '@typescript-eslint/naming-convention': 'off',
+      '@typescript-eslint/no-explicit-any': 'off',
+      'react/prop-types': 'off'
+    }
+  },
+  {
+    files: ['*.spec.*'],
+    ...jestPlugin,
+    rules: {
+      ...jestPlugin.rules,
+      'no-console': 'off',
+      'jest/prefer-snapshot-hint': 'off',
+      'jsx-a11y/label-has-associated-control': 'off',
+      'react/button-has-type': 'off',
+      'react/jsx-key': 'off'
+    }
+  },
+  {
+    files: ['*.stories.tsx', 'packages/*/demo/**/*'],
+    rules: {
+      'react/button-has-type': 'off',
+      'react/no-array-index-key': 'off',
+      'func-name-matching': 'off'
+    }
+  }
+];
```

Comment:
```suggestion
/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import prettierConfig from 'eslint-config-prettier';
import config from '@zendeskgarden/eslint-config';
import noticePlugin from '@zendeskgarden/eslint-config/plugins/notice.js';
import reactPlugin from '@zendeskgarden/eslint-config/plugins/react.js';
import typescriptPlugin from '@zendeskgarden/eslint-config/plugins/typescript.js';
import jestPlugin from '@zendeskgarden/eslint-config/plugins/jest.js';

const typescriptRules = {
  ...typescriptPlugin.rules,
  '@typescript-eslint/explicit-function-return-type': 'off',
  '@typescript-eslint/explicit-module-boundary-types': 'off',
  '@typescript-eslint/naming-convention': 'off',
  '@typescript-eslint/no-explicit-any': 'off',
  'n/no-unsupported-features/es-builtins': ['error', { version: '>=16.9.0' }],
  'react/prop-types': 'off'
};

export default [
  ...config,
  noticePlugin,
  reactPlugin,
  prettierConfig,
  {
    ignores: ['**/dist']
  },
  {
    rules: {
      'sort-imports': 'off',
      'react/jsx-no-useless-fragment': 'off'
    }
  },
  {
    files: ['packages/*/src/**/*.{ts,tsx}'],
    ignores: ['packages/.template/**/*.{ts,tsx}'],
    ...typescriptPlugin,
    rules: typescriptRules
  },
  {
    files: ['packages/*/src/**/*.spec.{ts,tsx}'],
    ignores: ['packages/.template/**/*.spec.{ts,tsx}'],
    ...typescriptPlugin,
    ...jestPlugin,
    rules: {
      ...typescriptRules,
      ...jestPlugin.rules,
      'no-console': 'off',
      'jest/prefer-snapshot-hint': 'off',
      'jsx-a11y/label-has-associated-control': 'off',
      'react/button-has-type': 'off',
      'react/jsx-key': 'off'
    }
  },
  {
    files: ['packages/*/demo/**/*.{ts,tsx}'],
    ignores: ['packages/.template/demo/**/*.{ts,tsx}'],
    ...typescriptPlugin,
    rules: {
      ...typescriptRules,
      'func-name-matching': 'off',
      'react/button-has-type': 'off',
      'react/no-array-index-key': 'off'
    }
  }
];
```

It's important to remember that there is no more "cascade" per se. I disabled rules and inserted code violations in order to test that the new flat config is working as expected. The updated config above will result in about 40 new v9 rule violations that should be addressed in code. Use `npx eslint --inspect-config` with "Test matching with filepath..." input to make sure that the expected rules are being applied to the right files.

Reply:
So if I'm understanding correctly, we need to re-apply the same set of rules to each new block if the extensions are the same as a previous rule block, even if we need to make slight variations? Why eslint. 😭

Reply:
The test file path feature is so helpful. At least they gave us _something_ in response to making the config more convoluted. lol

Before your suggestion
<img width="1199" alt="Screenshot 2024-07-19 at 10 14 41 AM" src="https://github.com/user-attachments/assets/99b1bdf5-de4c-40ce-8184-1475468ecade">

And after
<img width="1210" alt="Screenshot 2024-07-19 at 10 14 49 AM" src="https://github.com/user-attachments/assets/52970181-fb65-4f7d-97b4-bd18c1aea463">